pkg/internal/codegen/parse/index.go

```diff
@@ -183,61 +115,47 @@ type resourceTags struct {
 	ShortName string
 }
 
-// ParseResourceTag parses the tags in a "+resource=" comment into a resourceTags struct
-func parseResourceTag(tag string) resourceTags {
-	result := resourceTags{}
+//parseResourceAnnotationValue is a helper function to extract resource annotation.
+func parseResourceAnnotationValue(tag string, count int) (resourceTags, int, error) {
+	res := resourceTags{}
 	for _, elem := range strings.Split(tag, ",") {
-		kv := strings.Split(elem, "=")
-		if len(kv) != 2 {
-			log.Fatalf("// +kubebuilder:resource: tags must be key value pairs.  Expected "+
-				"keys [path=<subresourcepath>] "+
+		key, value, err := general.ParseKV(elem)
+		if err != nil {
+			return resourceTags{}, count, fmt.Errorf("// +kubebuilder:resource: tags must be key value pairs.  Expected "+
+				"keys [path=<resourcepath>] "+
 				"Got string: [%s]", tag)
 		}
-		value := kv[1]
-		switch kv[0] {
-		//case "rest":
-		//	result.REST = value
+		switch key {
 		case "path":
-			result.Resource = value
-		//case "strategy":
-		//	result.Strategy = value
+			res.Resource = value
 		case "shortName":
-			result.ShortName = value
+			res.ShortName = value
+		default:
+			return resourceTags{}, count, fmt.Errorf("The given input %s is invalid", value)
 		}
 	}
-	return result
+	count++
```

Comment:
in current form, count is just tracking if function was able to parse any resource annotation successfully or not which is also indicated by the returned error. So caller can decide this on the error basis. I don't see the need for count unless I am missing something ?

Reply:
So Instead of using boolean, I was keeping the count of resource annotations and if it exceeds beyond one per API type then I was throwing an error. So logic wise this code is doing the same thing what you proposed. Having said, it would be easier to keep one boolean variable and toggle its value based on certain conditions. I will change that logic right away.